componentified-todo-list/maryam/src/AddNewItem.js

```diff
@@ -0,0 +1,68 @@
+import React from 'react';
+import {
+  makeStyles,
+  ListItem,
+  IconButton,
+  ListItemSecondaryAction,
+  TextField
+} from '@material-ui/core';
+import {Add, ArrowDownward, ArrowUpward} from '@material-ui/icons'
+// ----------------------------------------------------------------
+const useStyles = makeStyles((theme) => ({
+  form: {
+    '& > *': {
+      margin: theme.spacing(1),
+    },
+  },
+}));
+// ----------------------------------------------------------------
+const AddNewItem = ({todoItems, setTodoItems}) => {
+    const classes = useStyles();
+    let newItem = {};
```

Comment:
__[HIGH]__  
This is not the right way to do it in react and may cause serious bugs. Use a state for the text field value and then create your `newItem` inside of each `addXClickHandler`s:
```jsx
const [textfieldValue, setTextfieldValue] = useState('');
...
const addUpClickHandler = () => {
  const newItem = { id: Date.now(), check: false, title: textfieldValue };
  setToDoItems([newItem, ...todoItems]);
};
...
<TextField
  ...
  onChange={e => setTextfieldValue(e.target.value)}
  ...
/>
...
```